### PR TITLE
Add parseCookieName, parseCookieValue methods

### PR DESCRIPTION
**Pull request by flotwig**
I wanted to validate cookie names and values separately, so I updated the code to allow for it. I split the existing regex in half and updated the existing code to use the new functions. All tests pass.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -26,18 +27,37 @@ const stripCookieHeader = header => {
...
+		return null;
+	}
+	return valueMatch[1] === undefined ?
+		valueMatch[2] :
+		valueMatch[1];
+};
+
 const parseCookiePair = cookiePair => {
-	const match = COOKIE_PAIR.exec(cookiePair);
+	const parts = cookiePair.split('=', 2);
```
Comment (charmander) [inline index.js]: I can’t check right now, but I think this will allow invalid cookie pairs to be parsed when they contain more than one `=` because of the unusual way `split` works in JavaScript. (Also, `=` is valid in quoted values.) Try splitting manually with `indexOf`?

File: index.js [hunk 2]
```diff
@@ -26,18 +27,37 @@ const stripCookieHeader = header => {
 	return header.substring(start, end + 1);
 };
 
+const parseCookieName = cookieName => {
+	const nameMatch = COOKIE_NAME.exec(cookieName);
```
Comment (charmander) [inline index.js]: The name is the entire thing, so the group can be removed from the `COOKIE_NAME` regex and this function can be simplified to

```js
cookieName.test(cookieName) ?
	cookieName :
	null
```

… which is a bit of an unusual implementation. Maybe it should export `isCookieName` instead, even though it’s less uniform?

**Review by charmander — CHANGES_REQUESTED** [review]
Thanks for contributing back your changes!

File: index.js (hunk 1 above)
Comment (flotwig) [reply index.js]: Hmm, if I'm reading this right, a valid `cookieName` can't contain an `=`, so this should always split a cookie pair into name and value. 

Do you see an edge case if a user provides some pathological input? I thought `parseCookiePair('name"="foo"')` might not work, but I added a test for it, just returns `null`. I'm kinda struggling to come up with a case where this doesn't work as expected.

This is the same as doing 
```js
const i = cookiePair.indexOf('='); 
const part0 = cookiePair.slice(0, i); 
const part1 = cookiePair.slice(i);
```
, just less verbose, I think.

File: index.js (hunk 2 above)
Comment (flotwig) [reply index.js]: Good point, cleaned that up.

File: index.js (hunk 1 above)
Comment (charmander) [reply index.js]: `.split('=', 2)` is like `.split('=').slice(0, 2)`, so I expect the problematic cases to be `a=b=c` (shouldn’t parse) and `a="b=c"` (should parse).

Comment (flotwig) [reply index.js]: Oh... wow... I never realized that JS continues to split past the `limit` parameter... :grimacing: 

Okay, fixed that. `a="b=c"` parses as expected. `a=b=c` parses to the same value as `a="b=c"`, which I believe is correct, since the unquoted cookie value is allowed to contain equals signs (according to my understanding). Pushed up some unit tests for these cases, lmk if that looks good.

Comment (charmander) [reply index.js]: Oh, you’re right, `a=b=c` should parse. Are you pushing the changes to the right place? The pull request doesn’t seem to have updated.

Comment (flotwig) [reply index.js]: Been pushing to the wrong branch, check now.

**Review by charmander — APPROVED** [review]

**Comment by charmander** [comment]
Thank you again! I’ll publish 3.1.0 soon.

**Comment by charmander** [comment]
Published 3.0.1 with a bugfix and 3.1.0 including it.

**Comment by flotwig** [comment]
Thanks!
